### DISCUSSION TITLE
Adhoc filter with resolved maps

### DISCUSSION THREAD

**Discussion by fionera** [grafana/clickhouse-datasource#612] [Ideas]
Currently when you try to set an adhoc filter you can only select between all different map states. Resolving the map to a full key would be easier to use

**Comment by krokwen** [comment]
Same, it would be very useful with OTEL

It would be nice to have an AdHoc filter over for example LogAttributes instead of whole-map json-like giant string in the selection.
In most cases the AdHoc filters are useless with OTEL due to attributes storage in maps

Configuration proposal:
`clickhouse_adhoc_query = 'otel_db.otel_logs.LogAttributes[]'`
unfortunately i have no idea what to propose for custom select query for map keys :(
and then the filter applies like `WHERE otel_db.otel_logs.LogAttributes['${filterSelectionKey}'] = '${filterSelectionValue}'`
